skins/oasis/modules/PageHeaderController.class.php

```diff
@@ -605,6 +605,24 @@ public function executeHubs( $params ) {
 		wfProfileOut( __METHOD__ );
 	}
 
+	public function executeAddNewPageButton() {
+		$this->shouldDisplay = false;
+
+		// That's when we show the number of pages on a wiki
+		if ( $this->pageExists() && !$this->wg->Title->isSpecial( 'Videos' ) ) {
+			$this->shouldDisplay = true;
+			$href = SpecialPage::getTitleFor( 'CreatePage' )->getLocalURL();
+
+			wfRunHooks( 'PageHeaderAfterAddNewPageButton', [ &$href ] );
+
+			$this->href = $href;
+		}
+	}
+
+	private function pageExists() {
+		return !empty( $this->wg->Title ) && $this->wg->Title->exists();
+	}
+
 	static function onArticleSaveComplete( &$article, &$user, $text, $summary,
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) This function has 11 parameters, which is greater than the 7 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS107)